### DISCUSSION TITLE
What would be the proper syntax to include a file with custom variables?

### DISCUSSION THREAD

**Discussion by Andre601** [fralau/mkdocs-macros-plugin#104] [Q&A]
From my research so far have I found out that I could use the `{% include <file> %}` syntax to include a separate file from the defined includes directory.

My question now is, if I can provide values to said syntax which would replace specific values within the targeted file.

I'll give a quick example of what my goal is:
I have a file called `img_table.md` which contains the following markdown (Note that I'm not sure about the syntax here regarding the values I want to have replaced)
```markdown
| Image | Description |
| --- | --- |
| ![{{ image }}]({{ image }}) | {{ description }} |
```
I would now like to use this file in `{% include img_table.md %}` and have the `{{ image }}` and `{{ description }}` syntaxes replaced with whatever values I provide in the include.

Is this doable? I'm fairly new to jinja2, macros and python in general and I'm not confident enough to try it completely on my own...

**Comment by fralau** [comment]
I would say _in principle_: yes, the `include` directive imports the context. 

See the [relevant paragraph](https://jinja.palletsprojects.com/en/3.0.x/templates/#import-context-behavior) in the official Jinja2 documentation

**Reply by Andre601** [reply]
Not quite sure how this answers my question here... I want to use the markdown file in other files, while replacing any placeholders in it with matching values.

Like I want to turn
```markdown
| Image | Description |
| --- | --- |
| ![{{ image }}]({{ image }}) | {{ description }} |

```
into
```markdown
| Image | Description |
| --- | --- |
| ![/assets/example.jpg](/assets/example.jpg) | An example image |

```
by providing the values in the `include`. Sorry if that wasn't clear.

**Reply by fralau** [reply]
Yes, if `image` and `description` are variable in the included file, that would work?

**Reply by Andre601** [reply]
Still wondering about the actual syntax of `include` tho. Like is it something like `{% include 'img_table.md' image='/assets/example.jph' description='An example image' %}`?

I would like to avoid working with YAML frontmatter in the files for multiple reasons.

**Reply by fralau** [reply]
I'm not sure I fully understand what you want to achieve. Why are you trying to define the variables in the `include` directive? What does the included file contain?

The way you do an assignment ([see the doc](https://jinja.palletsprojects.com/en/3.0.x/templates/#assignments)) is as follows:

```hmtl
{% set image='/assets/example.jph' %}
{% set description='An example image' %}
```

**Reply by fralau** [reply]
@Andre601 Was this discussion point solved for you?

**Reply by Andre601** [reply]
Not quite.

Right now do I have a `{% include 'tables/table.md' %}` which pulls in a file containing a table and I set certain values to insert into the frontmatter.

As an example:
```markdown
---
table:
  title: Some title
  description: Some description
  other: Another value
---

# File

{% include 'tables/table.md' %}
```

What I would like to do is include/add the table into the file and have the values set in that placeholder... But I lack the jinja2 knowledge to really understand how I would achieve this...

**Comment by fralau** [comment]
@Andre601 The documentation on `{% include ...}` is contained in that [paragraph of the Jinja2 documentation](https://jinja.palletsprojects.com/en/3.0.x/templates/#include).

**Reply by Andre601** [reply]
So essentially, the `include` would not be what I'm looking for here...

Because I want to implement a template where I define values in the pattern (`{% <something> <options and values...> %}` and it would return the Markdown file with the provided values inserted... See previous comments for all the context...

**Reply by fralau** [reply]
@Andre601  How did things go? Did you solve this issue to your satisfaction?

**Reply by Andre601** [reply]
I've stopped working on my project, so I no longer use this anymore.

**Reply by fralau** [reply]
OK and all the best on your endeavors 👍

**Comment by redtide** [comment]
If could be useful for someone else reading this: since using Jekyll (and now porting the project(s) to MKDocs) I use(d) to include HTML tables mixed with Liquid/Jinja template, loading variables from YAML/JSON files.
